src/App.tsx

```diff
@@ -1,18 +1,34 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const ITEMS_PER_PAGE_OPTIONS = [3, 5, 10, 20];
+
+const generateItems = (count: number) => getNumbers(1, count).map((n) => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const handlePerPage = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
+  };
+
+  const items = generateItems(42);
+  const total = items.length;
+  const fromPage = (currentPage - 1) * perPage;
+  const toPage = currentPage * perPage > total ? total
+    : currentPage * perPage;
+  const visibleItems = items.slice(fromPage, toPage);
```

Comment:
you can move some of this outside the component and some move to useEffect

Reply:
I don't know Hook 'useEffect' yet. Because it'll be in a future topic.